src/js/components/Legenda/index.js

```diff
@@ -33,9 +37,53 @@ const legendas = [
     },
 ];
 
-export function encontraLegenda(label) {
-
+function encontraLegenda(label) {
     const legendaEncontrada = legendas.find(legenda => legenda.label === label);
 
     return legendaEncontrada && legendaEncontrada.legenda;
-}
\ No newline at end of file
+}
+
+export default class Legenda extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            labels: props.labels,
+        };
+    }
+
+    componentWillReceiveProps(props) {
+        this.setState({
+            labels: props.labels,
+        });
+    }
+
+    render() {
+        const { labels } = this.state;
+
+        return (
+            <Table>
+                <thead>
+                    <tr>
+                        <th>Sigla</th>
+                        <th>Legenda</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {labels.map((label, index) => {
```

Comment:
can we create a separated render function?
passing the callback directly yo map we are creating a new function in each iteration